### PR TITLE
Slf4j related fixes - testing branch (community)

### PR DESCRIPTION
**Pull request by chibenwa**
Integration tests for https://github.com/apache/james-project/pull/76

### CODE REVIEW

File: mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java [hunk 1]
```diff
@@ -134,10 +136,28 @@ public String getMailetInfo() {
             ByteArrayInputStream inputStream = new ByteArrayInputStream(icsContent);
             return Stream.of(Pair.of(key, builder.build(inputStream)));
         } catch (IOException e) {
-            LOGGER.error("Error while reading input: " + icsContent, e);
+            if (LOGGER.isErrorEnabled()) {
+                String icsContentAsString;
+                try {
+                    icsContentAsString = new String(icsContent, Charset.forName("UTF-8"));
+                } catch (Throwable t) {
```
Comment (rouazana) [inline mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java]: s/Throwable/Exception/ ?

File: mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java [hunk 2]
```diff
@@ -134,10 +136,28 @@ public String getMailetInfo() {
...
+                LOGGER.error("Error while reading input: " + icsContentAsString, e);
+            }
             return Stream.of();
         } catch (ParserException e) {
-            LOGGER.error("Error while parsing ICal object: " + icsContent, e);
+            if (LOGGER.isErrorEnabled()) {
+                String icsContentAsString;
+                try {
+                    icsContentAsString = new String(icsContent, Charset.forName("UTF-8"));
+                } catch (Throwable t) {
```
Comment (rouazana) [inline mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java]: idem?

File: server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SendMailHandler.java [hunk 3]
```diff
@@ -77,8 +77,7 @@ public HookResult onMessage(SMTPSession session, Mail mail) {
                 recipientString = theRecipients.toString();
             }
             if (LOGGER.isInfoEnabled()) {
-                String infoBuffer = "Successfully spooled mail " + mail.getName() + " from " + mail.getSender() + " on " + session.getRemoteAddress().getAddress().toString() + " for " + recipientString;
-                LOGGER.info(infoBuffer.toString());
+                LOGGER.info("Successfully spooled mail " + mail.getName() + " from " + mail.getSender() + " on " + session.getRemoteAddress().getAddress().toString() + " for " + recipientString);
```
Comment (rouazana) [inline server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SendMailHandler.java]: why no parametrized login?

File: server/protocols/fetchmail/src/main/java/org/apache/james/fetchmail/FetchMail.java [hunk 4]
```diff
@@ -850,14 +844,10 @@ protected void setSessionParameters(HierarchicalConfiguration configuration) {
...
             for (HierarchicalConfiguration propConf : allProperties) {
-                properties.setProperty(propConf.getString("[@name]"), propConf.getString("[@value]"));
-                if (LOGGER.isDebugEnabled()) {
-                    StringBuilder messageBuffer = new StringBuilder("Set property name: ");
-                    messageBuffer.append(propConf.getString("[@name]"));
-                    messageBuffer.append(" to: ");
-                    messageBuffer.append(propConf.getString("[@value]"));
-                    LOGGER.debug(messageBuffer.toString());
-                }
+                final String nameProp = propConf.getString("[@name]");
```
Comment (rouazana) [inline server/protocols/fetchmail/src/main/java/org/apache/james/fetchmail/FetchMail.java]: no final on local variable?

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/LogMessageTest.java [hunk 5]
```diff
@@ -61,6 +62,7 @@
     @Before
     public void setup() {
         logger = mock(Logger.class);
+        when(logger.isInfoEnabled()).thenReturn(true);
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/LogMessageTest.java]: this commit "update unit tests" should not be a separated commit but a fixup, else it means some other commits where not building

File: mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java (hunk 1 above)
Comment (chibenwa) [reply mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java]: Fixed already (I did that remark to the contributor...)

File: mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java (hunk 2 above)
Comment (chibenwa) [reply mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java]: Idem...

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/LogMessageTest.java (hunk 5 above)
Comment (chibenwa) [reply mailet/standard/src/test/java/org/apache/james/transport/mailets/LogMessageTest.java]: I'll rebase it. Sorry, correcting commit order for other people is always a pain. I will have a try.

File: server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SendMailHandler.java (hunk 3 above)
Comment (chibenwa) [reply server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SendMailHandler.java]: Outdated!

**Pull request merged**